src/PostgREST/App.hs

```diff
@@ -291,9 +291,12 @@ handleRead headersOnly identifier context@RequestContext{..} = do
             )
           ]
           ++ contentTypeHeaders context
+        rsOrErrBody = if status == HTTP.status416
+          then Error.errorPayload (Error.ApiRequestError ApiRequestTypes.InvalidRange)
+          else LBS.fromStrict rsBody
```

Comment:
I see the 416 error body on GET and RPC but POST/PATCH also have some range logic, would it make sense to handle the 416 in the other methods too?

Reply:
I'm a bit confused about the [RFC](https://httpwg.org/specs/rfc9110.html#field.range) for `Range`:

> A server MUST ignore a Range header field received with a request method that is unrecognized or for which range handling is not defined. For this specification, [GET](https://httpwg.org/specs/rfc9110.html#GET) is the only method for which range handling is defined.

I interpret it as if we allow Range for PATCH, for instance, we do not need to ignore it, right? Then I'd agree that the same 416 error logic should be applied for these methods. 

---

On the other hand, the [416 status error](https://httpwg.org/specs/rfc9110.html#status.416) says:

> The 416 ([Range](https://httpwg.org/specs/rfc9110.html#field.range) Not Satisfiable) status code indicates that the set of ranges in the request's Range header field ([Section 14.2](https://httpwg.org/specs/rfc9110.html#field.range)) has been rejected...

Would this mean that the 416 error raised by  `offset` in the query string is not accurate since an error with `Range` is expected but no such header is present? The new tests I added would not be OK, then.

Reply:
I think the problem stems from using the same logic for `limit/offset` and `Range`. One has postgresql semantics which don't err while the HTTP semantic can err.

That being said, it's useful for web clients to get an error. Also from the [416 status error](https://httpwg.org/specs/rfc9110.html#status.416):

> or because the client has requested an excessive number of small or overlapping ranges (a potential denial of service attack).

Which is interesting because that could be extrapolated to: reply with an error whenever we have a `db-max-rows=1000` and a client requests a `limit=2000`. Right now we don't err and just prefer the 1000 limit.

---

I think the PR is good as it is, we can open a new issue for deciding what to do with the Range semantics on `limit/offset`.